milestone-5/backend/routes/recipe.ts

```diff
@@ -0,0 +1,68 @@
+// recipe routes
+import Recipe from "../models/recipeSchema";
+import express, { Request, Response, Router } from "express";
+
+const router: Router = express.Router();
+
+// get all recipes
+router.get("/", async (req: Request, res: Response) => {
+  const recipes = await Recipe.find({});
+  res.send(recipes);
+});
+
+// get specific recipe
+router.get("/:name", async (req: Request, res: Response) => {
+  const { name } = req.params;
+  const recipe = await Recipe.findOne({
+    name: name,
+  });
+  res.send(recipe);
+});
+
+// create new recipe
+router.post("/", async (req: Request, res: Response) => {
+  await Recipe.create(req.body);
+  res.send("Recipe created");
+});
+
+// update recipe
+router.put("/:name", async (req: Request, res: Response) => {
+  await Recipe.findOneAndUpdate({ name: req.params.name }, req.body);
+  res.send("Recipe updated");
+});
```

Comment:
you could add more error checking to your routes in order to protect you from annoying bugs down the line. try wrapping things in a try/catch block